sdk/python/kfp/compiler/compiler.py

```diff
@@ -1006,8 +1006,10 @@ def _validate_workflow(workflow: dict):
   if argo_path:
     result = subprocess.run([argo_path, 'lint', '/dev/stdin'], input=yaml_text.encode('utf-8'), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     if result.returncode:
-      raise RuntimeError(
-        '''Internal compiler error: Compiler has produced Argo-incompatible workflow.
+      err = result.stderr.decode('utf-8')
+      if dsl.RUN_ID_PLACEHOLDER not in err:
```

Comment:
Does Argo still complain if you use `{{workflow.labels.runid}}`?

Reply:
I'm not sure I understand the question.

I believe Argo will always complain for variables that don't exist in the CR definition when running `argo lint`.
That's the case here as well. Since the label is added by the apiserver (long after compilation time), `argo lint` raises error.